hyperion-shared-preferences/src/main/java/com/willowtreeapps/hyperion/sharedpreferences/list/SharedPreferencesListAdapter.java

```diff
@@ -29,11 +31,28 @@ public SharedPreferencesFileViewHolder onCreateViewHolder(@NonNull ViewGroup par
 
     @Override
     public void onBindViewHolder(@NonNull SharedPreferencesFileViewHolder holder, int position) {
-        holder.bind(preferences.get(position));
+        String pref = isFilterApplied ? filteredPreferences.get(position) : preferences.get(position);
+        holder.bind(pref);
     }
 
     @Override
     public int getItemCount() {
-        return preferences.size();
+        return isFilterApplied ? filteredPreferences.size() : preferences.size();
     }
+
+    void filter(String query) {
+        String filterQuery = query.toLowerCase();
```

Comment:
Is it possible for `query` to ever be `null`? Want to make sure we're avoiding possible NPEs.